containers/ecr-viewer/src/app/components/Filters.tsx

```diff
@@ -271,67 +286,108 @@ const FilterReportableConditions = () => {
  * - Updates the browser's query string when the filter is applied.
  */
 const FilterByDate = () => {
-  const { searchParams, updateQueryParam } = useQueryParam();
-
+  const today = new Date().toLocaleDateString("en-CA");
+  const { searchParams, deleteQueryParam, updateQueryParam, pushQueryUpdate } =
+    useQueryParam();
   const [filterDateOption, setFilterDateOption] =
     useState<string>(DEFAULT_DATE_RANGE);
+  const [startDate, setStartDate] = useState<string>("");
+  const [endDate, setEndDate] = useState<string>("");
   const isFilterDateDefault = filterDateOption === DEFAULT_DATE_RANGE;
 
-  const handleDateOptionChange = (
-    event: React.ChangeEvent<HTMLInputElement>,
-  ) => {
-    // Set option
-    const { value } = event.target;
-    setFilterDateOption(value);
-  };
-
   const resetFilterDate = () => {
-    const queryDateRange = searchParams.get("dateRange");
-    if (!queryDateRange) {
-      setFilterDateOption(DEFAULT_DATE_RANGE);
-    } else if (queryDateRange !== filterDateOption) {
+    const queryDateRange = searchParams.get(ParamName.DateRange) || "";
+    const queryCustomDates = searchParams.get(ParamName.Dates) || "";
+
+    if (isValidParamDates(queryDateRange, queryCustomDates)) {
       setFilterDateOption(queryDateRange);
+      const [startDate, endDate] = queryCustomDates?.split("|");
+      setStartDate(startDate);
+      setEndDate(endDate);
+    } else {
+      setFilterDateOption(DEFAULT_DATE_RANGE);
+      setStartDate("");
+      setEndDate("");
     }
   };
 
   // on mount, make sure the filters match the search params
   useEffect(resetFilterDate, []);
 
+  const submitHandler = () => {
+    if (filterDateOption === CustomDateRangeOption) {
+      if (!endDate) {
+        setEndDate(today);
+      }
+      const datesParam = `${startDate}|${endDate || today}`;
```

Comment:
```suggestion
      const actualEndDate = endDate || today;
      setEndDate(actualEndDate);
      const datesParam = `${startDate}|${actualEndDate}`;
```
maaaaybe this is a bit better for readability/maintainability? but it's marginal, so take or leave as you see fit